### DISCUSSION TITLE
OK to support fewer .NET Framework TFMs?

### DISCUSSION THREAD

**Discussion by rgwood** [Kinnara/ModernWpf#386] [Ideas]
# Summary

Right now, projects in the ModernWPF repo target 5 different .NET Framework TFMs:  `net45`, `net452`, `net46`, `net462`, `net48`

To simplify and speed up building ModernWPF I think it might be worth supporting fewer .NET Framework TFMs. I'd be happy to make this this change if others agree.

# Details

[Here are the official support dates for the .NET Framework](https://docs.microsoft.com/en-us/lifecycle/products/microsoft-net-framework). `net45` is already unsupported, `net452` and `net46` will become unsupported on 2022-04-26.

Personally I would vote to standardize on .NET 4.8 only. If that's too drastic,  `net462` and `net48`?

Project | .NET Framework TFMs
------------ | -------------
ModernWpf.Controls\ModernWpf.Controls.csproj | net45; net462
ModernWpf.MahApps\ModernWpf.MahApps.csproj | net452; net46; net462; net47
ModernWpf.SampleApp\ModernWpf.SampleApp.csproj | net45; net48
ModernWpf\ModernWpf.csproj | net45; net462
samples\FluentRibbonSample\FluentRibbonSample.csproj | net48
samples\MahAppsSample\MahAppsSample.csproj | net48
samples\MultiThreadingSample\MultiThreadingSample.csproj | net48
samples\PowerShellSample\PowerShellSample.csproj | net45
samples\SamplesCommon\SamplesCommon.csproj | net45; net462
test\ModernWpfTestApp\ModernWpfTestApp.csproj | net48
test\TestAppUtils\TestAppUtils.csproj | net48
tools\WinUIResourcesConverter\WinUIResourcesConverter.csproj | net48

**Comment by chucker** [comment]
Personally, I’d prefer leaving (technically, _adding_) 4.7.2 and 4.8.

Here’s the case for 4.7.2:

* I still see a single-digit percentage of customers who do not have 4.8
* 4.7.2 is the first release that correctly supports .NET Standard 2.0

**Reply by rgwood** [reply]
That sounds like a great compromise @chucker. If @Kinnara is supportive I'll submit a PR.
